docs/blogs/IntroGatewayAPI/GettingCreative.md

```diff
@@ -0,0 +1,294 @@
+Welcome to the foo application!
+```
+
+Next, let's test header matching:
+```shell
+$ curl -sH "Host: http.example.com" -H "x-app: bar" $INGRESS_HOST
+Welcome to the bar application!
+```
+
+You’ve successfully configured Gateway API to match traffic based on path, query parameters, and headers.
+
+### TLS Termination with Gateway API
+In OpenShift, we have the concept of [termination types](https://docs.openshift.com/container-platform/4.11/networking/routes/secured-routes.html) 
+to achieve different forms of secured routing. In this example, we will explore how to create an edge terminated route with Gateway API.
+
+An edge route in OpenShift terminates TLS encryption at the Ingress Controller before forwarding traffic to the
+destination pod:
+```yaml
+apiVersion: v1
+kind: Route
+metadata:
+  name: http-foo
+  namespace: demo-app
+spec:
+  host: www.edge.example.com
+  to:
+    kind: Service
+    name: foo-app
+  port:
+    targetPort: 8080
+tls:
+  termination: edge
+  insecureEdgeTerminationPolicy: Redirect
+  certificate: <CERT>
+  key: <KEY>
+path: /foo
+```
+In Gateway API, we can translate this concept to terminating the TLS encryption at Gateway before forwarding traffic to
+the HTTPRoute.
+
+First, we will need a certificate for configuring the TLS termination. Acquire a certificate with the common name that
+you want to use from your CA.  
+
+> For testing purposes, you can create your own CA and serving certificate as follows:\
+> $ mkdir certs\
+> $ openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj '/O=MyCompany/CN=example.com' -keyout certs/ca.key -out certs/ca.crt\
+> $ openssl req -out certs/edge.csr -newkey rsa:2048 -nodes -keyout certs/edge.key -subj "/CN=*.example.com/O=MyCompany"\
+> $ openssl x509 -req -sha256 -days 365 -CA certs/ca.crt -CAkey certs/ca.key -set_serial 0 -in certs/edge.csr -out certs/edge.crt\
+
+Add the certificate as a secret in the demo-gateway namespace:
+```shell
+$ oc create -n demo-gateway secret tls edge-cert --key=certs/edge.key --cert=certs/edge.crt
+```
+Certificates are handled differently in Gateway API compared to OpenShift `Routes`. Gateway API uses references to 
+Kubernetes `secrets` whereas OpenShift Route API embeds the certificate and key strings in the `Route` object.
+
+Next, let’s configure our Gateway to terminate TLS. We will add an additional Listener with the same wildcard domain
+using port 443 to our existing Gateway from the example above:
+```yaml
+$ oc apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+  namespace: demo-gateway
+spec:
+  gatewayClassName: istio
+  listeners:
+  - name: http
+    hostname: "*.example.com"
+    port: 80
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: All
+  - name: edge
+    hostname: "*.example.com"
+    port: 443
+    protocol: HTTPS
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - name: edge-cert
+        group: ""
```

Comment:
It's weird to specify `group` and not `kind`.  Since they default to the empty string and "Secret", respectively, they can both be omitted in this example.